### DISCUSSION TITLE
Regarding the application for adding new inbound and outbound functions

### DISCUSSION THREAD

**Discussion by Fintinger** [PamCatten/hot-stuff#4] [Ideas]
The current function can add or delete items, but if you want to modify the number of an item and record the modification time, it is more troublesome, so I suggest adding a function specifically to increase or decrease the number of an item. And can generate corresponding records. Thanks!

**Comment by PamCatten** [comment]
First of all, thanks for your suggestion. I think I understand most of what you're saying, but let's make sure anyway. 

You'd like a function that, when you're editing an item, can be called to increase/decrease the quantity of the item, record the time of the change, and to generate a corresponding report?

I have a couple of clarifying questions: 
* What's the use case for storing when edits are made to the item? Is the time presented in the UI? If so, where?
* Does this function only make changes to item quantity, or should it change other things as well?
* What records / reports do you want generated with this function? Are these viewable in the app, or downloaded to the device?

**Reply by Fintinger** [reply]
Thank you very much for your reply.
- The first question is that you do not need to call this function when editing products. What I mean is whether you can add a separate function to add or subtract items. This function is at the same level as the function of editing products.
- The second question is, actually, I am a warehouse management worker. What I have to do is not to count the number of each item every time, but to frequently increase or decrease items, that is, ''warehouse in and out''. 

> My suggestion is that there is no need to make any more adjustments to the existing `Items` main page. You only need to add the two functions of adding and subtracting to the `Items details` page. When adding or subtracting, you can fill in the quantity, date and remarks. etc. As for the UI part, you can jump to a separate page on the` item details `page to specifically display information such as increase or decrease time, quantity, etc.

- The third question actually already has an answer, it just needs to be dedicated to additions and subtractions.
- The fourth question is, if the function of increasing or decreasing items can be added, then two charts need to be added to the "Home" page to show the increase or decrease of each item within a certain period of time.

If this is the case, then this software has basic "warehouse in and out" functionality.

Looking forward to hearing from you!

**Reply by PamCatten** [reply]
Fantastic! I think we're on the same page now. I think these features seem doable individually, but I need to spend some more time weighing whether all of them make sense to implement here. 

To be clear, that's not a no; But the original design of the app was for a tool to help people prepare for potential insurance claims, and there's already just so much that needs to be finished within that scope before adding on the additional work needed to make this an effective tool for warehouse operations.

**Comment by Fintinger** [comment]
Thank you for your feedback.I'm already looking forward to it！
